src/VisualStudio/Core/Def/Implementation/PullMemberUp/VisualStudioPullMemberUpService.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.  
+
+using System.Composition;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeRefactorings.PullMemberUp;
+using Microsoft.CodeAnalysis.CodeRefactorings.PullMemberUp.Dialog;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.VisualStudio.Language.Intellisense;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.PullMemberUp
+{
+    [ExportWorkspaceService(typeof(IPullMemberUpOptionsService), ServiceLayer.Host), Shared]
+    internal class VisualStudioPullMemberUpService : IPullMemberUpOptionsService
+    {
+        private readonly IGlyphService _glyphService;
+
+        private PullMemberUpViewModel ViewModel { get; set; }
+
+        private AnalysisResult MembersInfo { get; set; }
+
+        [ImportingConstructor]
+        public VisualStudioPullMemberUpService(IGlyphService glyphService)
+        {
+            _glyphService = glyphService;
+        }
+
+        public PullMemberDialogResult GetPullTargetAndMembers(
```

Comment:
having a service talk about a 'DialogResult' feels really strange.  This service shouldn't expose anything to caller that indicates that there was a dialog involved.